ecl/hql/hqlexpr.cpp

```diff
@@ -1345,6 +1345,13 @@ extern HQL_API IPropertyTree * queryArchiveEntry(IPropertyTree * archive, const
 
 //---------------------------------------------------------------------------------------------------------------------
 
+HqlLookupContext::~HqlLookupContext()
+{
+    //If timings were not accumulated for this lookup context, accumulate child timings in the parent
+    if (container)
+        container->childCycles += childCycles;
```

Comment:
HqlLookupContext::reportTiming() updates container->childCycles with the elapsedCycles.  However, here it is updating the container->childCycles with this object's 'childCycles', rather than this object's elapsedCycle - this doesn't seem to be consistent.   It could make use of startCyles to work out elapsedCycles and then update the Container (startCycles could be reset at the end of reportTiming to avoid duplicating the stat).

Reply:
You only want to include in the child time, the time that is actually reported by a child.  So if the time is reported, the elapsed time is deducted, otherwise the time of any reported children is deducted.